nano/node/bootstrap/bootstrap_legacy.cpp

```diff
@@ -223,10 +223,10 @@ void nano::bootstrap_attempt_legacy::run ()
 			condition.wait (lock, [&stopped = stopped, &pulling = pulling] { return stopped || pulling == 0; });
 		}
 
-		// Flushing may resolve forks which can add more pulls
-		lock.unlock ();
-		node->block_processor.flush ();
-		lock.lock ();
+		while (!stopped && !node->active.empty ())
```

Comment:
If we are in operating in saturation mode, which we have to assume we will be, then this will mean that the legacy bootstrap will never try to do a bootstrap? I think it can't be a binary yes or no, if the AEC is busy/ful the bootstrap should be discouraged but never stopped completely.

Reply:
Looking at this, I made an incorrect chang by looking at the AEC rather than at the emptiness of the block_processor. I changed the implementation to check the block_processor is empty rather than the AEC is empty.

Reply:
But even then, my comment still holds, under saturation, will then block processor ever be empty?
And if it is never empty, will the bootstrap never kick in?